app/notifications/process_notifications.py

```diff
@@ -127,9 +127,16 @@ def send_notification_to_queue(notification, research_mode, queue=None):
         queue = QueueNames.RESEARCH_MODE
 
     if notification.notification_type == SMS_TYPE:
+        deliver_task = provider_tasks.deliver_sms
+        if send_to_providers._sms_send_on_pinpoint(
+            notification.notification_type,
+            notification.reply_to_text
+        ):
+            deliver_task = provider_tasks.deliver_throttled_sms
+            if not queue:
+                queue = QueueNames.SEND_THROTTLED_SMS
```

Comment:
Different task and queue is the SMS is sent with a dedicated phone number

Reply:
This makes sense codewise but takes a minute to understand the order of how things get assigned in which conditions - perhaps a quick comment explaining the logic?